sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -159,12 +171,13 @@ case class AlterTableRenameCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     DDLUtils.verifyAlterTableType(catalog, oldName, isView)
-    // If this is a temp view, just rename the view.
-    // Otherwise, if this is a real table, we also need to uncache and invalidate the table.
-    val isTemporary = catalog.isTemporaryTable(oldName)
-    if (isTemporary) {
-      catalog.renameTable(oldName, newName)
-    } else {
+
+    // If the old table name contains database part, we should rename a metastore table directly,
+    // otherwise, try to rename a temp view first, if that not exists, rename a metastore table.
+    val renameMetastoreTable =
+      oldName.database.isDefined || !catalog.renameTempView(oldName.table, newName)
```

Comment:
see https://github.com/apache/spark/pull/14962#discussion_r77808532

I'd like to avoid breaking existing behaviours here.